src/tests/roc_sndio/target_sndfile/test_sndfile_sink.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_core/heap_arena.h"
+#include "roc_core/temp_file.h"
+#include "roc_sndio/sndfile_sink.h"
+
+namespace roc {
+namespace sndio {
+
+namespace {
+
+enum { FrameSize = 500, SampleRate = 44100, ChMask = 0x3 };
+
+core::HeapArena arena;
+
+} // namespace
+
+TEST_GROUP(sndfile_sink) {
```

Comment:
I guess these tests are mostly (or completely) identical for all file backends.

It would be a big burden to maintain a copy for each backend, especially when we'll add more.

Can we instead replace tests for sox + test for sndfile with common backend-independent tests? Each test could query registered backends from BackendMap, select those which support files, and test each of them.

We use similar approach in other tests:

* https://github.com/roc-streaming/roc-toolkit/blob/master/src/tests/roc_fec/test_encoder_decoder.cpp#L124
* https://github.com/roc-streaming/roc-toolkit/blob/master/src/tests/roc_fec/test_writer_reader.cpp#L227
* https://github.com/roc-streaming/roc-toolkit/blob/master/src/tests/roc_audio/test_resampler.cpp#L303

Reply:
And I think test_pump should use the same approach.